index.js

```diff
@@ -215,26 +215,31 @@ function init (ssb, config) {
    */
 
   /* Tangle: auto-add tangles.group info to all private-group messages */
-  const getGroupTangle = GetGroupTangle(ssb, keystore)
+  const getGroupTangle = GetGroupTangle(ssb, keystore, 'group')
+  const getMembersTangle = GetGroupTangle(ssb, keystore, 'members')
   ssb.publish.hook(function (publish, args) {
     const [content, cb] = args
     if (!content.recps) return publish.apply(this, args)
 
     if (!isGroup(content.recps[0])) return publish.apply(this, args)
 
     onKeystoreReady(() => {
-      getGroupTangle(content.recps[0], (err, tangle) => {
-        // NOTE there are two ways an err can occur in getGroupTangle
-        // 1. recps is not a groupId
-        // 2. unknown groupId,
-
-        // Rather than cb(err) here we we pass it on to boxers to see if an err is needed
-        if (err) return publish.apply(this, args)
-
-        set(content, 'tangles.group', tangle)
-        tanglePrune(content) // prune the group tangle down if needed
+      if (!keystore.group.has(content.recps[0])) return cb(Error('unknown groupId'))
+
+      getGroupTangle(content.recps[0], (err, groupTangle) => {
+        if (err) return cb(Error("Couldn't get group tangle", { cause: err }))
+        getMembersTangle(content.recps[0], (err, membersTangle) => {
```

Comment:
you should check the `type` before wasting cycles building membership tangle - it could be you're already "done"

Reply:
yeah it's just annoying writing conditional CBs. but i'll do that

Reply:
done